### DISCUSSION TITLE
$(TargetPath) is empty is PostBuidEvent in Bundle project (Wix 4.0.1)

### DISCUSSION THREAD

**Discussion by mtissington** [wixtoolset/issues#7618] [Questions] [answered]
I know there are numerous posts on StackExchange about `$(TargetPath)` being empty on older versions of Visual Studio.
I'm using VS 2022 and this is only happening in new Bundle project.

I have read some older comment to use `!(TargetPath)` but that does not get resolved.

What am I missing to have `$(TargetPath) & $(TargetDir)` not be empty on my PostBuidlEvent ?

```
<Project Sdk="WixToolset.Sdk/4.0.1">
  <PropertyGroup>
    <OutputType>Bundle</OutputType>
    <OutputName>ActiveWords Setup</OutputName>
    <PostBuildEvent>for /F "tokens=1 delims==" %%a in ('$(SolutionDir)sigcheck -n -nobanner "$(TargetPath)"') do set VERSION = %%a 
xcopy "$(SolutionDir)$(ConfigurationName)Symbols\*.*" "$(TargetDir)\%VERSION%\" /D
copy "$(TargetPath)" "$(TargetDir)\%VERSION%\"
copy "$(TargetPdbPath)" "$(TargetDir)\%VERSION%\"
tar -caf %VERSION%.zip %VERSION%
rd /s /q %VERSION%</PostBuildEvent>
    <RunPostBuildEvent>OnBuildSuccess</RunPostBuildEvent>
  </PropertyGroup>
  <ItemGroup>
    <Content Include="logo.png" />
    <Content Include="logoSide.png" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="WixToolset.Bal.wixext" Version="4.0.1" />
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\Setup\Setup.wixproj" />
    <ProjectReference Include="..\WixBAFunctions\WixBAFunctions.vcxproj" />
  </ItemGroup>
</Project>
```

**Comment by mtissington** [comment]
Any ideas about this or is it a bug?

**Comment by robmen** [comment, marked as answer]
Others have already hit this, solutions discussed here (I think) and HeatWave has special case to migrate it.

You need to explicitly import the SDK.props/targets and place your usage of TargetDir after the last import. This is an MSBuild behavior.

PostBuildEvent is an anachronism in Sdk-style projects, recommendation is to move to explicit Target instead.

**Reply by mtissington** [reply]
Thanks for the hint - but I'm not sure I understand - the project already requires the sdk.
But move to "explicit Target" - any examples?

**Reply by barnson** [reply]
https://wixtoolset.org/docs/tools/msbuild/ has a link right near the top of the page.

**Comment by mtissington** [comment]
Thanks. I found this in an earlier post [https://github.com/orgs/wixtoolset/discussions/7393](https://github.com/orgs/wixtoolset/discussions/7393)

But this introduces a different issue - I had overridden the output filename (different to the project name) and when I import the sdks, the output is the project name - so where do I now set the output file name?

**Comment by mtissington** [comment]
For anyone finding this here is an example of a working project file. Of note, 
`<Import Project="Sdk.targets" Sdk="WixToolset.Sdk" Version="4.0.1" />`
MUST come AFTER `OutputType` and `OutputName`

```
<Project>
  <Import Project="Sdk.props" Sdk="WixToolset.Sdk" Version="4.0.1" />
  <PropertyGroup>
    <OutputType>Bundle</OutputType>
    <OutputName>ActiveWords Setup</OutputName>
    <PostBuildEvent>for /F "tokens=1 delims==" %%a in ('$(SolutionDir)sigcheck -n -nobanner "$(TargetDir)ActiveWords Setup.exe"') do set VERSION=%%a
echo "$(TargetDir)%VERSION%\"
xcopy "$(SolutionDir)$(Configuration)\Symbols\*.*" "$(TargetDir)%VERSION%\" /D

copy "$(TargetDir)ActiveWords Setup.exe" "$(TargetDir)%VERSION%\"
copy "$(TargetDir)ActiveWords Setup.wixpdb" "$(TargetDir)%VERSION%\"

tar -caf "%VERSION%.zip" "%VERSION%"
rd /s /q "%VERSION%"</PostBuildEvent>
    <RunPostBuildEvent>OnBuildSuccess</RunPostBuildEvent>
  </PropertyGroup>
  <ItemGroup>
    <Content Include="logo.png" />
    <Content Include="logoSide.png" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="WixToolset.Bal.wixext" Version="4.0.1" />
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\Setup\Setup.wixproj" />
    <ProjectReference Include="..\WixBAFunctions\WixBAFunctions.vcxproj" />
  </ItemGroup>
  <Import Project="Sdk.targets" Sdk="WixToolset.Sdk" Version="4.0.1" />
</Project>
```
